backend/lcfs/utils/spreadsheet_builder.py

```diff
@@ -0,0 +1,143 @@
+                    for cell in row:
+                        cell.number_format = '#,##0'
+                        cell.alignment = styles.Alignment(horizontal='left')
+
+        self._apply_excel_styling(writer, sheet)
+
+    def _apply_excel_styling(self, writer, sheet):
+        worksheet = writer.sheets[sheet["sheet_name"]]
+
+        # Apply no border, non-bold font, and left alignment to header cells
+        for cell in worksheet[1]:
+            cell.border = styles.Border(
+                left=styles.Side(style=None),
+                right=styles.Side(style=None),
+                top=styles.Side(style=None),
+                bottom=styles.Side(style=None)
+            )
+            cell.font = styles.Font(bold=False)
+            cell.alignment = styles.Alignment(horizontal='left')
+
+        if sheet["styles"].get("bold_headers"):
+            for cell in worksheet[1]:
+                cell.font = styles.Font(bold=True)
+
+        for i, width in enumerate(sheet["styles"].get("column_widths", []), start=1):
+            worksheet.column_dimensions[chr(64 + i)].width = width
+
+    def _write_xls(self, output):
+        book = xlwt.Workbook()
+
+        for sheet_data in self.sheets_data:
+            self._write_sheet_to_xls(book, sheet_data)
+        book.save(output)
+
+    def _write_sheet_to_xls(self, book, sheet_data):
+        sheet = book.add_sheet(sheet_data["sheet_name"])
+
+        # Styles for headers
+        bold_style = xlwt.XFStyle()
+        font = xlwt.Font()
+        font.bold = True
+        bold_style.font = font
+        default_style = xlwt.XFStyle()
+
+        # Style for left-aligned numbers
+        left_aligned_num_style = xlwt.XFStyle()
+        left_aligned_num_style.num_format_str = '#,##0'
+        alignment = xlwt.Alignment()
+        alignment.horz = xlwt.Alignment.HORZ_LEFT
+        left_aligned_num_style.alignment = alignment
+
+        # Apply bold style for headers if required
+        header_style = bold_style if sheet_data["styles"].get("bold_headers") else default_style
+
+        for col_index, column in enumerate(sheet_data["columns"]):
+            sheet.write(0, col_index, column, header_style)
+            if "column_widths" in sheet_data["styles"] and len(sheet_data["styles"]["column_widths"]) > col_index:
+                width = sheet_data["styles"]["column_widths"][col_index]
+                sheet.col(col_index).width = 256 * width
+
+        for row_index, row in enumerate(sheet_data["rows"], start=1):
+            for col_index, value in enumerate(row):
+                # Apply left-aligned style for numbers, default style for others
+                cell_style = left_aligned_num_style if isinstance(value, (int, float)) else xlwt.XFStyle()
+                sheet.write(row_index, col_index, value, cell_style)
+
+    def _write_csv(self, output):
+        if self.sheets_data:
```

Comment:
Are you thinking we wouldn't have CSVs with more than one sheet_data? I'm fine with this but add a comment saying so.

Reply:
I forgot to add a note for this! I've added a comment to mention it. Thank you, Alex!